src/main/scala/akka/stream/alpakka/xlsx/CellReference.scala

```diff
@@ -7,68 +7,68 @@ import akka.stream.alpakka.xml.Attribute
 import scala.annotation.tailrec
 import scala.util.Try
 
-case class CellReference(name: String, colNum: Int, rowNum: Int)
+final case class CellReference(fullName: String, columnIndex: Int, rowIndex: Int)
 
 object CellReference {
+  private final val BaseLettersNum = ('A' to 'Z').size
+  private final val AsciiCodeA     = 'A'
+  private final val AsciiCodeZ     = 'Z'
 
-  private val CELL_REF = """(\s+"""
 
-  private def convertColStringToIndex(ref: String): Int = {
-    @tailrec
-    def solver(refArray: List[Char], index: Int, retval: Int): Int = {
-      refArray.headOption match {
-        case Some(c) =>
-          if (index != 0) throw new IllegalArgumentException("Bad col ref format '" + ref + "'")
-          else solver(refArray.tail, index + 1, retval * 26 + c - 65 + 1)
-        case None => retval
-      }
+  private[xlsx] def parseReference(attributes: List[Attribute]): Option[CellReference] = {
+    def splitCellReference(reference: String) = {
+      val referenceLength       = reference.length
+      val referenceColumn       = reference.filterNot(c => c >= '0' && c <= '9')
+      val referenceColumnLength = referenceColumn.length
+
+      if (referenceLength == 0 || referenceColumnLength == 0 || referenceColumnLength == referenceLength) None
+      else Some((reference, referenceColumn, reference.substring(referenceColumnLength)))
```

Comment:
Can we be sure `.substring` never throws?

Reply:
Yes, we can. Please note that `referenceColumn.length` is both non-negative and not greater than `reference.length` (since it is obtained by `reference` filtering).